ethcoder/ethcoder.go

```diff
@@ -24,6 +20,28 @@ func AbiCoder(argTypes []string, argValues []interface{}) ([]byte, error) {
 		}
 		args = append(args, abi.Argument{Type: abiType})
 	}
+	return args, nil
+}
+
+func AbiDecoder(argTypes []string, input []byte, argValues []interface{}) error {
```

Comment:
nice work on this.

any thoughts if this would work / be better api?

```go
args := []interface{}{&big.Int{},&big.Int{})
// or even, args := make([]*big.Int, 2)
err := AbiDecoder([]string{"uint256","uint256"}, packedData, args)

//..

// then, impl of AbiDecoder could be..
func AbiDecoder(argTypes []string, input []byte, v interface{}) error {
  // same as before, but we accept "v" as interface{} type and so we can decode into any 
  // data structure, similar to how go's json/encoding works
}
```

the reason for the above changes is so that the data types will already be of the type expected by the caller, without having to pass []interface{} then have to cast it after the decoding

Reply:
If now sure if we can choose the decoding output, I think it will always be an array of interfaces, so its a matter of declaring the array in the call or outside the call. My point for doing it outside is that you wont be using that array anymore, this is, after AbiDecoder, you wont be using 'args' anymore, you will likely have use in args pointers to fields in a struct.